string/string.h

```diff
@@ -0,0 +1,297 @@
+/**
+ * Library for embedded strings
+ *
+ * @author Aaron Chan
+ */
+
+#ifndef STRING_H
+#define STRING_H
+
+#include <cstddef>
+
+const char NULL_TERMINATOR = '\0';
```

Comment:
probably make this static

Reply:
or maybe use constexpr?

Reply:
Dont think its necessary if both classes have access to it?

Reply:
it's not really good practice to have a global declaration in a header file, every file that includes this will be exporting a symbol for "NULL_TERMINATOR", if for some reason someone has a definition for the same name anywhere in their project the linker will have two symbols with the same name and it won't link. Making it static explicitly means it won't export a symbol so no linker issues

Reply:
Ah got it